syntax = "proto3";

/* namespace */
package hello;

// import "google/protobuf/empty.proto";

// The request message containing the user's name.
message HelloRequest {
    string name = 1;
}
  
// The response message containing the greetings
message HelloReply {
    string message = 1;
}

// The greeting service definition.
service Greeter {
    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloReply);
    // Sends another greeting
    rpc SayHelloAgain (HelloRequest) returns (HelloReply);
    // Sends a greeting stream
    rpc SayHelloOften (HelloRequest) returns (stream HelloReply);
}